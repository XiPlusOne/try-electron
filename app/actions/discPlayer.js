// @flow
export const DISC_START_SPIN = 'DISC_START_SPIN';
export const DISC_STOP_SPIN = 'DISC_STOP_SPIN';
export const ROD_ON = 'ROD_ON';
export const ROD_OFF = 'ROD_OFF';
export const CHANGE_DISC = 'CHANGE_DISC';
export const CHANGE_SOUNDTRACK = 'CHANGE_SOUNDTRACK';
export const CHANGE_TITLE = 'CHANGE_TITLE';
export const CHANGE_DURATION = 'CHANGE_DURATION';
export const CHANGE_CURRENT_TIME = 'CHANGE_CURRENT_TIME';

export function discStartSpin() {
  return {
    type: DISC_START_SPIN
  };
}

export function discStopSpin() {
  return {
    type: DISC_STOP_SPIN
  };
}

export function rodOn() {
  return {
    type: ROD_ON
  };
}

export function rodOff() {
  return {
    type: ROD_OFF
  };
}

export function changeDisc(discPic: string) {
  return {
    type: CHANGE_DISC,
    data: discPic
  };
}

export function changeSoundTrack(soundTrack: string) {
  return {
    type: CHANGE_SOUNDTRACK,
    data: soundTrack
  };
}

export function changeTitle(title: string) {
  return {
    type: CHANGE_TITLE,
    data: title
  };
}

export function changeDuration(duration: number) {
  return {
    type: CHANGE_DURATION,
    data: duration
  };
}

export function changeCurrentTime(currentTime: number) {
  return {
    type: CHANGE_CURRENT_TIME,
    data: currentTime
  };
}
