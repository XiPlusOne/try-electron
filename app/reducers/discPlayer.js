// @flow
import { combineReducers } from 'redux';
import {
  DISC_START_SPIN,
  DISC_STOP_SPIN,
  ROD_ON,
  ROD_OFF
} from '../actions/discPlayer';
import type { Action } from './types';

function isDiscSpinning(state: boolean = false, action: Action) {
  switch (action.type) {
    case DISC_START_SPIN:
      return true;
    case DISC_STOP_SPIN:
      return false;
    default:
      return state;
  }
}

function isRodOn(state: boolean = false, action: Action) {
  switch (action.type) {
    case ROD_ON:
      return true;
    case ROD_OFF:
      return false;
    default:
      return state;
  }
}

export default combineReducers({
  isDiscSpinning,
  isRodOn
});
