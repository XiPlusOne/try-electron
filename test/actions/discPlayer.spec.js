import * as actions from '../../app/actions/discPlayer';

describe('actions', () => {
  it('should discStartSpin should create discStartSpin action', () => {
    expect(actions.discStartSpin()).toMatchSnapshot();
  });

  it('should discStopSpin should create discStopSpin action', () => {
    expect(actions.discStopSpin()).toMatchSnapshot();
  });

  it('should rodOn should create rodOn action', () => {
    expect(actions.rodOn()).toMatchSnapshot();
  });

  it('should rodOff should create rodOff action', () => {
    expect(actions.rodOff()).toMatchSnapshot();
  });
});
