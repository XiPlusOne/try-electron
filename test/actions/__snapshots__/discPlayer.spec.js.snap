// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should discStartSpin should create discStartSpin action 1`] = `
Object {
  "type": "DISC_START_SPIN",
}
`;

exports[`actions should discStopSpin should create discStopSpin action 1`] = `
Object {
  "type": "DISC_STOP_SPIN",
}
`;

exports[`actions should rodOff should create rodOff action 1`] = `
Object {
  "type": "ROD_OFF",
}
`;

exports[`actions should rodOn should create rodOn action 1`] = `
Object {
  "type": "ROD_ON",
}
`;
